src/sentry/snuba/outcomes.py

```diff
@@ -227,19 +230,26 @@ def __init__(
         if len(raw_fields) == 0:
             raise InvalidField('At least one "field" is required.')
         self.fields = {}
-        self.aggregations = []
         self.query: List[Any] = []  # not used but needed for compat with sessions logic
         start, end, rollup = get_constrained_date_range(query, allow_minute_resolution)
-        self.dataset = _outcomes_dataset(rollup)
+        self.dataset, self.match = _outcomes_dataset(rollup)
         self.rollup = rollup
         self.start = start
         self.end = end
-
+        self.conditions = [
+            Condition(Column("timestamp"), Op.GTE, start),
+            Condition(Column("timestamp"), Op.LT, end),
+        ]
+        self.select_params = []
         for key in raw_fields:
             if key not in COLUMN_MAP:
                 raise InvalidField(f'Invalid field: "{key}"')
             field = COLUMN_MAP[key]
-            self.aggregations.append(field.aggregation(self.dataset))
+            aggregation = field.aggregation(self.dataset)
```

Comment:
see previous comment, re: replacing aggregation on field with select_parameter method, or let me know if you want to talk through this